M2/Macaulay2/m2/programs.m2

```diff
@@ -42,15 +50,26 @@ getProgramPath = (name, cmds, opts) -> (
 	    if opts.Verbose == true and #prefixes > 1 then
 		print("  trying prefix \"" | prefix_1 |
 		    "\" for executables matching \"" | prefix_0 | "\"...");
+	    newOpts := if opts.MinimumVersion =!= null then
+		new OptionTable from apply(keys opts, key -> key =>
+		    if key === MinimumVersion then (opts#key_0,
+			pathToTry | addPrefix(opts#key_1, prefix))
+		    else opts#key)
```

Comment:
using `merge` here would be better

Reply:
Cool, I wasn't aware of `merge` -- I'll check it out.  It may be a moot point if I refactor `checkProgramPath` to deal with the path/prefix substitution as we wouldn't need to do this anyway.